torchtext/experimental/transforms.py

```diff
@@ -251,3 +251,56 @@ def to_ivalue(self):
             vector = self.vector.to_ivalue()
             return VectorTransform(vector)
         return self
+
+
+class PadTransform(nn.Module):
+    r"""Pad sequences
+
+    Args:
+        pad_id: the id for padding token
+        bos_token_id: the id for bos token (Default: None)
+        eos_token_id: the id for eos token (Default: None)
+        return_key_padding_mask: flag to output key_padding_mask as output (Defualt: True)
+
+    Example:
+        >>> pad_transform = PadTransform(2)
+        >>> seq_batch = [[5, 4, 5, 6, 7], [1, 3], [7, 5, 8]]
+        >>> pad_seq, key_padding_mask = pad_transform(seq_batch)
+        >>> jit_pad_transform = torch.jit.script(pad_transform)
+    """
+
+    def __init__(self, pad_id, bos_token_id=None, eos_token_id=None, return_key_padding_mask=True):
+        super(PadTransform, self).__init__()
+        self.pad_id = pad_id
+        self.bos_token_id = bos_token_id
+        self.eos_token_id = eos_token_id
+        self.return_key_padding_mask = return_key_padding_mask
+
+    @torch.jit.export
+    def forward(self, seq_batch: List[List[int]]) -> Tuple[torch.Tensor, Optional[Tensor]]:
```

Comment:
We have efficient padding layers including types of padding https://pytorch.org/docs/master/nn.html#padding-layers